# List of all field descriptions

## Kids First Dataservice Fields

### universal dataservice fields

{% docs kf_id %}
Unique id automatically assigned by the Kid's First DRC.
{% enddocs %}

{% docs created_at %}
Time of object creation.
{% enddocs %}

{% docs modified_at %}
Last time of object modification.
{% enddocs %}

{% docs uuid %}
A Universally Unique Identifier which is a 128-bit number used to uniquely identify information in Kids First Dataservice. This is automatically generated by Kids First Dataservice. Not actively used in any downstream operations.
{% enddocs %}

{% docs visible %}
Flags visbility of a record in Kids First Dataservice. Used in downstream operations to determine if data is eligble to be ingested into FHIR and released to the KF Portal. When set to false, data is not permitted to be released. When set to true, data can be released and used for downstream processes. Set by the data analyst at time of ingest and modified when eligible for release. Only for internal use.
{% enddocs %}

{% docs visibility_reason %}
Gives justification for the value in the visible column from a set of enums. Set by the data analyst at the time of ingest and modified as needed. Only for internal use.
{% enddocs %}

{% docs visibility_comment %}
Additional details for the visbility reason. Typically notes the associated jira ticket number when applicable. Set by the data analyst at the time of ingest and modified as needed. Only for internal use. 
{% enddocs %}

{% docs external_id %}
External ID or source entity ID used by contributor. In the case of genomic files, this is the full s3 path of a genomic file. Is only unique within a study, not across all studies.
{% enddocs %}

{% docs study_id %}
The Kids First assigned kf id that represents which study a participant, specimen, or genomic file are associated with. In the format, "SD_XXXXXXXX"
{% enddocs %}

{% docs investigator_id %}
The Kids First assigned kf id that represents the Principle Investigator of one study. In the format, "IG_XXXXXXXX"
{% enddocs %}

{% docs participant_id %}
The Kids First assigned kf id that represents a participant. In the format, "PT_XXXXXXXX"
{% enddocs %}

{% docs family_id %}
The Kids First assigned kf id that represents a family. In the format, "FM_XXXXXXXX"
{% enddocs %}

{% docs specimen_id %}
The Kids First assigned kf id that represents a biospecimen. In the format, "BS_XXXXXXXX"
{% enddocs %}

{% docs genomic_file_id %}
The Kids First assigned kf id that represents a genomic file. In the format, "GF_XXXXXXXX"
{% enddocs %}

{% docs sequencing_experiment_id %}
The Kids First assigned kf id that represents a sequencing experiment. In the format, "SE_XXXXXXXX"
{% enddocs %}

{% docs bsgf_id %}
The Kids First assigned kf id that represents the linkage between a specimen and a file. In the format, "BG_XXXXXXXX"
{% enddocs %}

{% docs segf_id %}
The Kids First assigned kf id that represents the linkage between a genomic file and a sequencing experiment. In the format, "SG_XXXXXXXX"
{% enddocs %}

{% docs diagnosis_id %}
The Kids First assigned kf id that represents a diagnosis. In the format, "DG_XXXXXXXX"
{% enddocs %}

{% docs phenotype_id %}
The Kids First assigned kf id that represents a phenotype. In the format, "PH_XXXXXXXX"
{% enddocs %}

{% docs age_at_event_days %}
Age at the time of event occurred in number of days since birth
{% enddocs %}

{% docs name %}
For investigator tables: the full name of the investigator. For study tables: the title of the study.
{% enddocs %}

### diagnosis fields

{% docs dewrangle_diagnosis_id %}
The dewrangle generated id for a diagnosis. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "DG_" with "cn-".
{% enddocs %}

{% docs source_text_diagnosis %}
The pathological diagnosis. Is typically the original diagnosis text received from the data contributors in clinical data submissions. For Kids First Datasets, is typically harmonized to match the text associated to the MONDO ID term.
{% enddocs %}

{% docs mondo_id_diagnosis %}
The ID of the term from the Monary Disease Ontology which represents a harmonized diagnosis. This ID is either provided to the Kids First DRC by the contributor or is selected by the Kids First DRC during curation of condition data.
{% enddocs %}

{% docs icd_id_diagnosis %}
The ID of the term from the International Classification of Diseases which represents harmonized diagnosis. Not actively populated.
{% enddocs %}

{% docs diagnosis_category%}
High level diagnosis categorization. Not actively populated.
{% enddocs %}

{% docs source_text_tumor_location %}
Only applicable for cancer studies. Represents the location of the tumor.
{% enddocs %}

{% docs uberon_id_tumor_location %}
The ID of the term from Uber anatomy ontology which represents harmonized anatomical ontologies. Not actively populated.
{% enddocs %}

### family fields

{% docs dewrangle_family_id %}
The dewrangle generated id for a family. This id is a lower-cased version of the corresponding kf id.
{% enddocs %}

{% docs family_type %}
Denotes type of family using a set of enums, such as proband only or trio. Not currently populated in Kids First dataservie, but is calculcated by the portal etl and displayed on the Kids First portal. 
{% enddocs %}

### genomic_file fields

{% docs dewrangle_genomic_file_id %}
The dewrangle generated id for a genomic file. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "GF_" with "dr-".
{% enddocs %}

{% docs is_harmonized %}
Denotes whether a genomic file is harmonized to standard outputs or if it is a raw unharmonized genomic file. When set to true, the file is harmonized. When set to false, the file is unharmonized. 
{% enddocs %}

{% docs reference_genome %}
Original reference genome of the unharmonized genomic files. Obtained from the sequencing center.
{% enddocs %}

{% docs controlled_access %}
Denotes whether a file is controlled or open access. When set to true, a file is controlled access and must have a corresponding controlled access acl/authz value. When set to false, a file is open access and must have the open access acl/authz value. All raw source files are by default controlled access. Harmonized genomic outputs controlled access may vary and are determined using a standard mapping file maintained by the bioinformatics team.
{% enddocs %}

{% docs availability %}
Indicates whether a file is available for immediate download, or is in cold storage. This value is obtained directly from AWS.
{% enddocs %}

{% docs paired_end %}
The direction of the read. Not applicable for every file type.
{% enddocs %}

{% docs data_type %}
A more specific denotation of the type of genomic file. 
{% enddocs %}

{% docs file_format %}
Denotes the file extension. 
{% enddocs %}

{% docs data_category %}
Inidicates the broad category of the type of data file. 
{% enddocs %}

{% docs workflow_tool %}
Specifies the specific tool within the workflow used to generate the file. For source files, this field should be NULL. These values are maintained by the bioinformatics team.
{% enddocs %}

{% docs workflow_type %}
Denotes the generic name for the workflow utilized in analysis. For source files, this field should be NULL. These values are maintained by the bioinformatics team.
{% enddocs %}

{% docs workflow_version %}
Indicates the major version of the workflow. For source files, this field should be NULL. These values are maintained by the bioinformatics team.
{% enddocs %}

{% docs workflow_endpoint %}
The endpoint name of the task outputs from CAVATICA task or the CAVATICA app output name. These values are maintained by the bioinformatics team.
{% enddocs %}

{% docs file_version_descriptor %}
Denotes whether a file is unharmonized, a previous verison or the latest version. 
{% enddocs %}

{% docs cavatica_file_id %}
Indicates the file ID in CAVATICA. Populated and maintained by the Kids First bioinformatics team.
{% enddocs %}

{% docs cavatica_volume %}
Indicates the CAVATICA volume ID, mediating cloud storage access. Popualted and maintained by the Kids First bioinformatics team.
{% enddocs %}

### investigator fields 

{% docs dewrangle_investigator_id %}
The dewrangle generated id for an investigator. This id is a lower-cased version of the corresponding kf id.
{% enddocs %}

{% docs institution %}
The name of the investigator's institution. 
{% enddocs %}

### participant fields 

{% docs alias_group_id %}
Not actively populated or used.
{% enddocs %}

{% docs dewrangle_participant_id %}
The dewrangle generated id for a participant. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "PT_" with "pt-".
{% enddocs %}

{% docs is_proband %}
Denotes whether participant is proband of study. When set to true, the participant is the proband of their family. When set to false, the participant is not the proband of their family.There are no restrictions on number of probands. A family can have zero or as many probands as designated by the Principle Investigator in the original clinical data submission.
{% enddocs %}

{% docs race %}
The race of the participant.
{% enddocs %}

{% docs ethnicity %}
The ethnicity of the participant. 
{% enddocs %}

{% docs gender %}
The genetic sex of the participant
{% enddocs %}

{% docs affected_status %}
Denotes whether a participant is affected by the disease or cancer focus of the study. When set to true, a participant has the disease/cancer. When set to false, a participant does not have the disease/cancer. Participants that have this value set to true, should have a correspnding condition record in the database.
{% enddocs %}

{% docs species %}
The species of the research particpant
{% enddocs %}

### phenotype fields 

{% docs dewrangle_phenotype_id %}
The dewrangle generated id for a phenotype. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "PH_" with "cn-".
{% enddocs %}

{% docs source_text_phenotype %}
Name given to phenotype by contributor. Is typically the original phenotype text received from the data contributors in clinical data submissions. For Kids First Datasets, is typically harmonized to match the text associated to the HPO ID term.
{% enddocs %}

{% docs hpo_id_phenotype %}
The ID of the term from Human Phenotype Ontology which represents a harmonized phenotype. This ID is either provided to the Kids First DRC by the contributor or is selected by the Kids First DRC during curation of condition data.
{% enddocs %}

{% docs observed %}
Denotes whether a phenotype is negative or positive
{% enddocs %}

{% docs snomed_id_phenotype %}
The ID of the term from Systematized Nomenclature of Medicine --Clinical Terms which encodes clinical terminology. Not actively populated.
{% enddocs %}

### sequencing experiment fields

{% docs dewrangle_sequencing_experiment_id %} 
The dewrangle generated id for a phenotype. This id is a lower-cased version of the corresponding kf id.
{% enddocs %}

{% docs sequencing_center_id %}
The kf_id of the sequencing center.
{% enddocs %}

{% docs experiment_date %}
Date of the sequencing experiment conducted. Not actively populated.
{% enddocs %}

{% docs experiment_strategy %}
Text term that represents the Library strategy.
{% enddocs %}

{% docs is_paired_end %}
Boolean term specifies whether reads have paired end.
{% enddocs %}

{% docs platform %}
Name of the platform used to obtain data.
{% enddocs %}
    
{% docs instrument_model %}
Text term that represents the model of instrument.
{% enddocs %}

{% docs library_name %}
Text term that represents the name of the library.
{% enddocs %}

{% docs library_strand %}
Text term that represents the library stranded-ness.
{% enddocs %}

{% docs library_prep %}
Text term that describes the library preparation method.
{% enddocs %}

{% docs library_selection %}
Text term that describes the library selection method.
{% enddocs %}

{% docs max_insert_size %}
Maximum size of the fragmented DNA.
{% enddocs %}

{% docs mean_insert_size %}
Mean size of the fragmented DNA.
{% enddocs %}

{% docs mean_depth %}
Mean depth or coverage describes the amount of sequence data that is available per position in the sequenced genome territory.
{% enddocs %}

{% docs total_reads %}
Total reads of the sequencing experiment.
{% enddocs %}

{% docs mean_read_length %}
Mean lenth of the reads.
{% enddocs %}

{% docs adapter_sequencing %}
Specifies the base sequence of the sequencing adapter. 
{% enddocs %}

{% docs is_adapter_trimmed %}
Indicates whether the FASTQ file has undergone adapter trimming.
{% enddocs %}

{% docs read_pair_number %}
Identifies the specific target capture kit utilized. Can be either the production-level kit name or a file path.
{% enddocs %}

{% docs target_capture_kit %}
Identifies the specific target capture kit utilized. Can be either the production-level kit name or a file path.
{% enddocs %}

{% docs acquisition_type %}
Data acquisition type.
{% enddocs %}

{% docs cdna_read %}
The type of read that contains the cDNA read.
{% enddocs %}

{% docs cdna_read_offset %}
The offset in sequence for the cDNA read.
{% enddocs %}

{% docs cell_barcode_offset %}
The offset in sequence of the cell identifying barcode.
{% enddocs %}

{% docs cell_barcode_read %}
The type of read that contains the cell barcode.
{% enddocs %}

{% docs cell_barcode_size %}
The size of the cell identifying barcode.
{% enddocs %}

{% docs chromatography_approach %}
Column type used for liquid chromatography.
{% enddocs %}

{% docs end_bias %}
The end of the cDNA molecule that is preferentially sequenced.
{% enddocs %}

{% docs enrichment_approach %}
Enrichment method used for phospho, ubiquitin,acetyl, or other enrichment.
{% enddocs %}

{% docs fraction_number %}
Number of fractions generated for sample.
{% enddocs %}

{% docs fractionation_approach %}
Method used to fractionate sample.
{% enddocs %}

{% docs ion_fragmentation %}
Reporter ion MS level.
{% enddocs %}

{% docs library_construction %}
The library construction method (including version) that was used
{% enddocs %}

{% docs mass_spec_rawfile_conversion %}
Method of converting raw spectra files to XML-based format.
{% enddocs %}

{% docs proteomics_experiment %}
The type of omics experiment that the sample was subject to.
{% enddocs %}

{% docs quantification_label_id %}
Specific label used or NA if label-free.
{% enddocs %}

{% docs quantification_labeling_method %}
Method used for labeling or NA if label-free.
{% enddocs %}

{% docs quantification_technique %}
Approach used for peptide quantification.
{% enddocs %}

{% docs sequencing_mode %}
The modes of sequencing technology.
{% enddocs %}

{% docs target_cell_number %}
The target number of cells per experiment or library.
{% enddocs %}

{% docs umi_barcode_offset %}
The offset in sequence of the UMI identifying barcode.
{% enddocs %}

{% docs umi_barcode_read %}
The type of read that contains the UMI barcode.
{% enddocs %}

{% docs umi_barcode_size %}
The size of the UMI identifying barcode.
{% enddocs %}

### specimen fields

{% docs dewrangle_specimen_id %}
The dewrangle generated id for a specimen. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "BS_" with "bs-".
{% enddocs %}

{% docs external_sample_id %}
Name given to sample by contributor.
{% enddocs %}

{% docs external_aliquot_id %}
Name given to aliquot by contributor.
{% enddocs %}

{% docs source_text_tissue_type %}
Description of the kind of biospecimen collected.
{% enddocs %}

{% docs composition %}
The cellular composition of the biospecimen.
{% enddocs %}

{% docs source_text_anatomical_site %}
The anatomical location of collection.
{% enddocs %}

{% docs source_text_tumor_descriptor %}
Disease present in the biospecimen.
{% enddocs %}

{% docs analyte_type %}
The molecular description of the aliquot.
{% enddocs %}

{% docs dbgap_consent_code %}
Consent classification code from dbgap.
{% enddocs %}

{% docs consent_type %}
Short name of consent.
{% enddocs %}

{% docs method_of_sample_procurement %}
The method used to procure the sample used to extract analyte(s).
{% enddocs %}

{% docs sample_id %}
The kf_id of the sample this specimen comes from.
{% enddocs %}

{% docs specimen_status %}
Whether the specimen is available for further processing in the biorepository.
{% enddocs %}

{% docs has_matched_normal_sample %}
Indicates if the biospecimen has a matched normal sample.
{% enddocs %}

{% docs shipment_origin %}
The original site of the aliquot.
{% enddocs %}

{% docs concentration_mg_per_ml %}
The concentration of the aliquot.
{% enddocs %}

{% docs volume_ul %}
The volume of the aliquot.
{% enddocs %}

{% docs shipment_date %}
The date the aliquot was shipped.
{% enddocs %}

{% docs uberon_id_anatomical_site %}
The ID of the term from Uber-anatomy ontology which represents harmonized anatomical ontologies.
{% enddocs %}

{% docs ncit_id_tissue_type %}
The ID term from the National CancerInstitute Thesaurus which represents a harmonized tissue_type.
{% enddocs %}

{% docs ncit_id_anatomical_site %}
The ID term from the NationalCancer Institute Thesaurus which represents a harmonized anatomical_site.
{% enddocs %}

{% docs spatial_descriptor %}
Ontology term that harmonizes thespatial concepts from Biological Spatial Ontology.
{% enddocs %}

{% docs preservation_method %}
Text term that represents the method used to preserve the sample.
{% enddocs %}

{% docs amount %}
The amount of the aliquot.
{% enddocs %}

{% docs amount_units %}
The units of the amount of the aliquot.
{% enddocs %}

{% docs cell_entity %}
The type of single cell entity.
{% enddocs %}

### study fields 
{% docs data_access_authority %}
Not actively populated.
{% enddocs %}

{% docs version %}
dbGaP version.
{% enddocs %}

{% docs short_name %}
Short name for study.
{% enddocs %}

{% docs attribution %}
Link to attribution prose provided by dbGaP.
{% enddocs %}

{% docs release_status %}
Release status of the study.
{% enddocs %}

{% docs dewrangle_study_id %}
The dewrangle generated id for a study. This id is a lower-cased version of the corresponding kf id, which replaces the pre-fix "SD_" with "sd-".
{% enddocs %}

{% docs short_code %}
Alphanumeric string to identify the study.
{% enddocs %}

{% docs domain %}
The diagnosis domain of the study (e.g. Cancer).
{% enddocs %}

{% docs program %}
The project/program the study is a part of.
{% enddocs %}

{% docs parent_study_id %}
KF ID of the parent study.
{% enddocs %}

{% docs biobank_email %}
Email for study biobank or source of specimens.
{% enddocs %}

{% docs biobank_name %}
Name of study biobank or source of specimens.
{% enddocs %}

{% docs biobank_request_instructions %}
Instructions for requesting specimens from study biobank.
{% enddocs %}

{% docs biobank_request_link  %}
Link to request specimens from study biobank.
{% enddocs %}


## Kids First Source Data Fields

### Clinical Data

{% docs family_relationship_target_participant_id %}
Participant ID of Proband of Family
{% enddocs %}

{% docs aliquot_id %}
Aliquot ID from sequencing manifest
{% enddocs %}

{% docs consent_group %}
Consent group of study
{% enddocs %}

{% docs age_at_sample %}
Age sample was collected from sequencing manifest
{% enddocs %}

{% docs sample_composition %}
Composition of specimen
{% enddocs %}

{% docs sample_tissue_type %}
Tissue type of collected specimen
{% enddocs %}

{% docs sex %}
Sex of pariticipant
{% enddocs %}

{% docs age_at_phenotype %}
Age of participant when phenotype was asserted
{% enddocs %}

{% docs vital_status %}
Vital status of participant
{% enddocs %}


### Broad Manifest 

{% docs collaborator_sample_id %}
Sample ID submitted by PI to sequencing center 
{% enddocs %}

{% docs collaborator_participant_id %}
Participant ID submitted by PI to sequencing center 
{% enddocs %}

{% docs cram_path %}
path of cram file from sequencing manifest
{% enddocs %}

{% docs crai_path %}
path of crai file from sequencing manifest
{% enddocs %}

{% docs md5_path %}
path of md5 file from sequencing manifest
{% enddocs %}

{% docs root_sample_id %}
Specimen sample ID from sequencing manifest
{% enddocs %}

### S3 Scrape

{% docs mean_coverage %}
mean coverage value from sequencing manifest
{% enddocs %}

{% docs bucket %}
s3 bucket for file
{% enddocs %}

{% docs key %}
s3 key for file
{% enddocs %}

{% docs last_modified %}
last time s3 file was modified
{% enddocs %}

{% docs etag %}
etag hash from s3 manifest
{% enddocs %}

{% docs size %}
size of file from s3 manifest
{% enddocs %}

{% docs storage_class %}
s3 storage class for file
{% enddocs %}


## Study Specific Fields

# SD_1NNS3K8V Fields